lldb/source/Core/SourceManager.cpp

```diff
@@ -300,6 +300,16 @@ size_t SourceManager::DisplaySourceLinesWithLineNumbersUsingLastFile(
         break;
       }
     }
+
+    Checksum checksum = last_file_sp->GetFileSpec().GetChecksum();
+    if (checksum && checksum != last_file_sp->GetChecksum()) {
+      llvm::call_once(last_file_sp->GetChecksumOnceFlag(), [&]() {
+        s->Printf("warning: source file checksum mismatch between the debug "
```

Comment:
Should we make a `DiagnosticReport` here ?

Reply:
I agree. Wherever we hardcode the string "error:" or "warning:" we should probably use a more high-level API.

Reply:
CC: @PortalPete

Reply:
We could use what I'm working on in my other PR, but we'd have to change this to pass the `CommandReturnObject` instead of just its `Stream` instance (via `&result..GetOutputStream()`).